src/hooks/useSettings.ts

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useMemo, useState } from "react";
+
+import { DEFAULT_SHOW_CHAT } from "settings";
+
+import { Settings } from "types/Firestore";
+import { SparkleSelector } from "types/SparkleSelector";
+
+import { isTruthy } from "utils/types";
+
+import { useFirestoreConnect } from "hooks/useFirestoreConnect";
+import { useSelector } from "hooks/useSelector";
+
+export const settingsSelector: SparkleSelector<Settings | undefined> = (
+  state
+) => state.firestore.data.settings;
+
+export interface UseSetingsReturnType {
+  areLoaded: boolean;
+  settings: Required<Settings>;
+}
+
+export const useSettings: () => UseSetingsReturnType = () => {
+  useFirestoreConnect([
+    { collection: "settings", doc: "customizations", storeAs: "settings" },
+  ]);
+
+  const settings = useSelector(settingsSelector);
+  const [areLoaded, setAreLoaded] = useState<boolean | undefined>(undefined);
+
+  useEffect(() => {
+    if (areLoaded === undefined) {
+      setAreLoaded(false);
+    } else {
+      setAreLoaded(true);
+    }
+    // disabling because we should be dependent only on settings here
+  }, [settings]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  return useMemo(
+    () => ({
+      areLoaded: isTruthy(areLoaded),
```

Comment:
```suggestion
      isSettingsLoaded: isLoaded(settings),
```

Reply:
haha, fixed 🙈 
the commit is here: https://github.com/sparkletown/sparkle/pull/2261/commits/e37f952af6f7c168cafd88fd5a8710c09d0bcfd6

Thank youuuuuu!! ❤️